sql/src/main/java/io/crate/execution/engine/aggregation/GroupBySingleNumberCollector.java

```diff
@@ -239,7 +249,13 @@ private void iter(Groups groups, Row row) {
     }
 
     private void addWithAccounting(Groups groups, Number key, Object[] newStates) {
-        ramAccounting.addBytes(groups.entryOverhead);
+        int mapSize = groups.statesByKey.size();
+        // If mapSize is a power of 2 then the map is going to grow by doubling its size.
+        // The first 4 elements have been calculated during the map construction.
+        if (mapSize >= 4 && (mapSize & (mapSize - 1)) == 0) {
+            ramAccounting.addBytes(mapSize * groups.entryOverhead);
```

Comment:
Shouldn't we only add bytes based on the number of "new" slots in the underlying arrays instead of always adding `mapSize * entryOverhead`

Reply:
But the `mapSize` new slots are created in the keys and values underlying arrays since the map is doubled.